app/services/workflow_executions/cancellation_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module WorkflowExecutions
+  # Service used to Cancel a WorkflowExecution
+  class CancellationService < BaseService
+    def initialize(workflow_execution, wes_connection, user = nil, params = {})
+      super(user, params)
+      @workflow_execution = workflow_execution
+      @wes_client = Integrations::Ga4ghWesApi::V1::Client.new(conn: wes_connection)
+    end
+
+    def execute
+      return false unless @workflow_execution.submitted?
+
+      # throws exception if failed
+      @wes_client.cancel_run(@workflow_execution.run_id)
```

Comment:
If an exeception is thrown, we should attach the exeception message to the @workflow_execution so that we can return the error. This service isn't going to be kicked off as a job correct? It will just be the user clicking a button to cancel which will fire off immediately?

Reply:
@ericenns Thoughts on this?

Reply:
All of the `wes_client` calls will throw appropriate exceptions from `lib/integrations/api_exceptions.rb`. I added the note in this PR but all of the other services that use the ga4gh wes api will throw exceptions on failure.

Reply:
Sounds good thanks. Is it possible to show the exception in the test `cancel unsubmitted workflow_execution`?

Reply:
Will add in the exception handling in a separate PR as this applies to all the services where the wes client is used